pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,212 @@
+    signal.signal(signal.SIGINT, _exit_handler)
+    if ui.ask_yes_no(constants.Strings.server_prompt.value):
+        domain = ui.ask_string(constants.Strings.server_ip.value)
+        if domain is None:
+            ui.fatal(constants.Strings.error_invalid_domain.value)
+    else:
+        domain = "play.pommerman.com:5050"
+    ui.info(
+        constants.Strings.server_connecting_p1.value, ui.yellow,
+        constants.Strings.server_connecting_p2.value, ui.reset,
+        constants.Strings.server_connecting_p3.value
+    )
+    network = Network(domain)
+    try:
+        status = network.server_status()
+    except Exception as e:
+        ui.fatal(e)
+    ui.info(
+        constants.Strings.server_connected.value, ui.yellow, constants.Strings.server_players.value,
+        str(status[0]) + ",", constants.Strings.server_matches.value, status[1]
+    )
+    intent(network)
+
+
+def intent(network):
+    """Description: This creates a prompt for the user where they can choose to:  
+    * Play a match  
+    * Create/Join a room  
+    * Replay a match  
+    * Exit the application  
+    Arguments:   
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    """
+    i = ui.ask_choice(
+        constants.Strings.intent.value, [
+            constants.Strings.intent_match.value, constants.Strings.intent_room.value,
+            constants.Strings.intent_replay.value, constants.Strings.intent_exit.value
+        ]
+    )
+    if i == constants.Strings.intent_match.value:
+        match(network)
+    elif i == constants.Strings.intent_room.value:
+        match(network, True)
+    elif i == constants.Strings.intent_replay.value:
+        replay(network)
+    elif i == constants.Strings.intent_exit.value:
+        exit(0)
+
+
+def match(
+    network, room=False, agent=False, ui_en=True
+):
+    """Description: This facilitates playing a match  
+    Arguments:  
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    * room: The room to be created/joined. If False, the public room will be \
+joined  
+    * agent: The class of the agent should be a derivative of BaseAgent  
+    * ui_en: If the UI is enabled or disabled (This also controls if exception \
+are raised or not)"""
+    if room:
+        room = str(ui.ask_string(constants.Strings.room_code.value))
+    sys.path.append(os.getcwd())
+    if not agent and ui_en:
+        agent = importlib.import_module(ui.ask_string(constants.Strings.match_import.value))
+        agent_class = ui.ask_string(constants.Strings.match_class_name.value)
+        if agent_class not in agent.__dir__():
+            if ui_en:
+                ui.fatal(constants.Strings.error_invalid_class.value)
+            raise Exception(constants.Exceptions.match_invalid_class.value)
+        agent = getattr(agent, agent_class)
+    if getattr(agent, "act") and ui_en:
+        ui.info(ui.green, constants.Strings.match_agent_success.value)
+    agent = agent()
+    if ui_en:
```

Comment:
Additionally, I recall saying somethign similar about the "get_agent" function hidden in the above. Please address that as well.

Reply:
It's pointless to have a 5 line function.

Reply:
I'm not sure what was up I think I was too tired so I wrote "Done" and didn't actually do that ?

Reply:
Where is `get_agent`?

Reply:
It's there somewhere in the file. It's called `_agent_prompt`.